main.tf

```
@@ -36,6 +36,11 @@ resource "azurerm_storage_account" "this" {
   }
 }
 
+resource "azurerm_advanced_threat_protection" "this" {
+  target_resource_id = azurerm_storage_account.this.id
```

Should be enabled for the SQL server 🙂 

```suggestion
  target_resource_id = azurerm_mssql_server.this.id
```